### PR TITLE
#18: Add code of conduct

### LINKED ISSUE #18: Add tab for rules and server guidelines (written as markdown parsed to HTML)

### PR DESCRIPTION
**Pull request by ianvln**
### Purpose of pull request

**Related Issue**: resolves #18 

Added a code of conduct page with content that was copy-pasted from a Google result.

I chose to go for the implementation using MDX template pages. If I need to update something, let me know!

### Pull request checklist

- [x] Branch and PR are named correctly
- [x] Updated relevant documentation
- [ ] Tested locally on the website
- [ ] Wrote unit tests for changes

### Testing instructions

N/A

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const withMDX = require('@next/mdx')({
+    extension: /\.mdx$/
+  })
+  
+  module.exports = withMDX({
+    pageExtensions: ['js', 'jsx', 'ts', 'tsx', 'md', 'mdx']
+  })
```
Comment (sirNikolai) [inline next.config.js:7]: nl

File: src/pages/code-of-conduct.mdx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+[INSERT CONTACT METHOD].
```
Comment (sirNikolai) [inline src/pages/code-of-conduct.mdx:72]: @brendacs I guess this would be server channel?

Comment (brendacs) [reply src/pages/code-of-conduct.mdx:72]: Linking to https://cscareerhub.com/contact would actually be better

**Review by brendacs — CHANGES_REQUESTED** [review]
I would prefer the code of conduct to be from our Discord server's rules in the #welcome channel. I'm also ok with having two - one for server rules and one for this code of conduct

**Comment by brendacs** [comment]
Great job with this! And MDX was a good choice. Just one other comment - make sure to use the `stage` branch as the base instead of `master`.

File: next.config.js (hunk 1 above)
Comment (ianvln) [reply next.config.js:7]: nl? 😄

**Comment by ianvln** [comment]
> I would prefer the code of conduct to be from our Discord server's rules in the #welcome channel. I'm also ok with having two - one for server rules and one for this code of conduct

I've added a separate page for the server rules

File: next.config.js (hunk 1 above)
Comment (brendacs) [reply next.config.js:7]: It means new line

**Review by brendacs — APPROVED** [review]
looks great!

**Pull request merged**